packages/react-native/src/Authenticator/common/FederatedProviderButtons/FederatedProviderButtons.tsx

```diff
@@ -1,21 +1,24 @@
 import React, { useMemo } from 'react';
 import { View } from 'react-native';
-import { capitalize } from '@aws-amplify/ui';
 
-import { icons } from '../../../assets';
+import { SocialProvider, authenticatorTextUtil } from '@aws-amplify/ui';
+import { useAuthenticator } from '@aws-amplify/ui-react-core';
+
 import { FederatedProviderButton } from '../FederatedProviderButton';
 import { Divider } from '../../../primitives';
-
+import { icons } from '../../../assets';
 import { styles } from './styles';
-import { FederatedProviderButtonsProps } from './types';
 
-export default function FederatedProviderButtons({
-  socialProviders,
-  toFederatedSignIn,
-}: FederatedProviderButtonsProps): JSX.Element | null {
+const { getSignInWithFederationText, getOrText } = authenticatorTextUtil;
+
+export default function FederatedProviderButtons(): JSX.Element | null {
+  const { route, socialProviders, toFederatedSignIn } = useAuthenticator(
+    ({ route, socialProviders }) => [route, socialProviders]
+  );
```

Comment:
Retrieve props from context via the `useAuthenticator` hook rather than prop drill. This break existing patterns in RNA but the alternative would mean that DefaultContent API has knowledge of these props, only needed for sign in and sign up. Anti-pattern aside, this feels wrong.